### PR TITLE
Starting to add new files, clean up, XE and XE7 test projects updated

### PR DESCRIPTION
**Pull request by rmcginty**
I'm going to go ahead and start this PR because I need to see if it checks against XE7.  I have added a few of the support files, associated tests, more whitespace cleanup, and a bug fix or two that move me closer to moving the rest of my bigger changes.  I do not yet have a CI that hits my GitHub repo, only my internal - trying to get that under control to better know that my repo is square prior to requesting a PR.

If the tests pan out with XE7, then I'll work on updating the addition project files to get them up to speed and add them to the PR.  Thinking of spending the time to automate this process though, as it is just too time consuming and error prone by hand for D2010 - 10.2, which I believe is like 12 project versions, times 3 projects (expert, tests, examples).

### CODE REVIEW

File: DUnitX.Utils.pas [hunk 1]
```diff
@@ -3332,13 +3324,40 @@ class function TListStringUtils.ToArray(const values: TList<string>): TArray<str
...
 end;
 
+class function TGUIDUtils.CleanGUIDString(const StrGUID: string; const RemoveDashes: boolean): string;
+begin
+  Result := TStrUtils.RemoveChars(StrGUID, '{}' + IfThen(RemoveDashes, '-'));
+end;
+
+class function TGUIDUtils.CreateGUID: TGUID;
+begin
+  SysUtils.CreateGUID(Result);
```
Comment (vincentparrett) [inline DUnitX.Utils.pas]: Namespcing issue..needs ifdef'd using ns for System.SysUtils

File: DUnitX.FileUtils.pas [hunk 2]
```diff
@@ -0,0 +1,185 @@
+{***************************************************************************}
```
Comment (vincentparrett) [inline DUnitX.FileUtils.pas]: Should use the same naming convention as exiting utils units. So should be DUnit.Utils.File.pas

**Comment by rmcginty** [comment]
Got it thanks - sorry I just realized that the cause of the failure isn't actually posted and you have to do that manually!  

I'll try to update the PR later today.  I also think I am almost there at getting my GitHub under Continua to build against XE and Berlin.  I figure it if builds against those two, it will at least cover the namespaces issues.

**Comment by rmcginty** [comment]
Also, it is normal that ALL my commits continue to show even though you've already merged those, then I merged yours back into mine?  The FILES part looks like I expected, but the COMMITS just seems to continue to grow with everything since I forked your repo again.  Am I missing or not doing something I should be?

**Comment by vincentparrett** [comment]
I have no idea why it's showing all those commits.. I am beginiing to suspect something is very wrong with this repo.

**Comment by vincentparrett** [comment]
Something isn't quite right with your fork/ local clone. Did you pull changes from this repo into your local clone before pushing to your fork?

https://help.github.com/articles/syncing-a-fork/

I'm no git guru, but google tells me you probably need to rebase - 

git rebase master
git push -f origin master

The -f (force) is needed because you would have already pushed these commits before. 

http://stackoverflow.com/questions/18323665/pull-request-on-github-showing-commits-rebased-from-master

http://stackoverflow.com/questions/16306012/github-pull-request-showing-commits-that-are-already-in-target-branch

**Comment by vincentparrett** [comment]
We're experiencing some issues with Continua CI and this repo at the moment, which seems to be related to closed pull requests that were never merged (and the forks deleted) causing github to generate auto merge commits on those pull requests every time we push a commit or merge a pull request. I've put in a support request to github.

**Comment by rmcginty** [comment]
I'm the first to admit that I am thoroughly confused about what is going on.  I approached this just like merging in Hg of two "related" repos, but I am beginning to see this is not the case.  I have not rebased - that is what I tried to do before deleting my previous repo because it totally went into the weeds.

What I've done is just:

1) Fork
2) Make changes, commit
3) PR
4) You merge
5) I pull and it makes me merge into mine , then I push to my repo
6) Go back to 2.

It doesn't seem like the merge you do reflects on my fork when I pull, which is odd to me.  I would've expected your merge to make my fork "converge" with the master, but it doesn't look like that on the graph.  Again, I'm super confused as to why this isn't easier, but perhaps it is just my lack of understanding how GitHub handles forks.  I read somewhere that they are technically clones and not actually related repos, so it doesn't work like "regular" Git, but I am not sure if that is accurate or not.

I'll read up on the links you provided and also ask a friend of mine that has much more experience with Git than I do...  Sorry for the trouble - I didn't realize it was so easy to get this out of whack!

**Comment by rmcginty** [comment]
Unless I'm reading the links wrong, it appears I'm not the only one that thinks this should automatically be happening and that GitHub has said this is by design.  I'll see if some of the fixes work on my local but I'm not pushing anything unless it looks exactly as I expect locally!

**Comment by vincentparrett** [comment]
DUnitX.Tests.Utils.Files.pas(55): error F1026: File not found: 'W:\CI_WS\Ws\798881\Source\DUnitX\DUnitX.FileUtils.dcu'

**Comment by rmcginty** [comment]
File is a reserved word, so the rename had to be DUnitX.Utils.Files.pas.  However, I didn't realize the other files where it was in use, so fixing that.

I did what the articles said, and it said everything was already fine - like it wouldn't let me rebase, so...  I don't know if the merge commit just prior to my last changed something or what...  Just going to finish this PR so we can get it closed and I'm going to get my CI building off the GitHub repo so I can stop pushing broken commits...

**Comment by vincentparrett** [comment]
DUnitX.Tests.Utils.Files.pas(86): error E2003: Undeclared identifier: 'FileAssert' 
DUnitX.Tests.Utils.Files.pas(86): error E2066: Missing operator or semicolon
DUnitXTest_XE7.dpr(78): error F2063: Could not compile used unit 'DUnitX.Tests.Utils.Files.pas'

**Comment by vincentparrett** [comment]
About time delphi unreserved File.. who still uses the old file stuff anyway?

**Comment by rmcginty** [comment]
I have no idea - I hope no one!  Ok, I didn't realize that FileAssert was in that test - giving this ONE more shot to pass checks then I'm going to just have to call it a night and see about getting my local going to find these before I commit.  I know you have better things to do than pasting compile errors here :)

**Comment by vincentparrett** [comment]
Thanks, all compiles and tests pass again.

**Comment by rmcginty** [comment]
Good deal - man I was pulling my hair there for a few.  Between GitHub and the side effects of diffing without being able to test - ugh!

As for GitHub though, if you notice, all my previous merges were basically "you merged vsoft" but the last one I did says that I merged INTO vsoft.  This makes my local repo graph look more like expect (they actually converge instead of continuously running parallel) but I guess we won't know until my next PR because it definitely did not allow me to do anything with the commits I've already made.  Full disclosure:  Someone who REALLY knows what they are doing might be able to fix it, but I'm not willing to put "force" on the end of anything at this point until I understand WHY I am doing it.  As I said though, hopefully I was just missing the merge into vsoft.

It is very confusing though, because I thought it was saying to try to merge to YOUR repo, which of course I don't have publish rights too, so I figured it was worth a shot.  I think what it actually means is, it merged my master with my local copy/history of your repo master locally and this part of it is what helps GitHub know that the repos are equal at that point.  That's my working theory anyway...  Guess we will see.

**Comment by rmcginty** [comment]
I'm about to add more changes to the PR (mostly due to updating the project files for consistency).  I am setting up Continua to pull from my repo and compile so, unless you hear otherwise, if the check fails, let me see if I can get my system to show me what is going on.  If it hit a dead end, I'll see what your side says, but HOPEFULLY I can avoid that from now on.

**Comment by vincentparrett** [comment]
Can you merge master into your repo again and update this PR? I know it says there are no conflicts but want to be sure before merging. Going to make an effort to clean up this repo, have not had much time to work on it lately.

**Comment by rmcginty** [comment]
Yeah, give me a day or so to get reacquainted with GitHub's merging and I'll get it knocked out!

**Comment by vincentparrett** [comment]
No problem.

**Comment by vincentparrett** [comment]
Closing this as there's been no activity in over a year.

**Pull request closed without merging**